Catalog/Examples/Networking/Japx/NetworkingJapxViewController.swift

```diff
@@ -0,0 +1,131 @@
+//
+
+import UIKit
+
+final class NetworkingJapxViewController: UIViewController {
+
+    // MARK: - Public properties -
+    
+    var presenter: NetworkingJapxPresenterInterface!
+    
+    // MARK: - Private properties -
+    
+    @IBOutlet private weak var _emailTextField: UITextField!
+    @IBOutlet private weak var _usernameTextField: UITextField!
+    @IBOutlet private weak var _passwordTextField: UITextField!
+    
+    @IBOutlet private weak var _createUserButton: UIButton!
+    @IBOutlet private weak var _getUserButton: UIButton!
+    @IBOutlet private weak var _updateUserButton: UIButton!
+    @IBOutlet private weak var _deleteUserButton: UIButton!
+    
+    @IBOutlet private weak var _scrollView: UIScrollView!
+    @IBOutlet private weak var _bottomConstraint: NSLayoutConstraint!
+    
+    private var _bottomHeight: CGFloat = 20
+    
+    // MARK: - Lifecycle -
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        updateView(with: .UserDoesNotExist)
+    }
+	
+    override func viewWillAppear(_ animated: Bool) {
+        _startObservingKeyboardEvents()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        _stopObservingKeyboardEvents()
+    }
+    
+    // MARK: - Button actions -
+    
+    @IBAction func createUserActionHandler(_ sender: Any) {
+        presenter.didPressCreate(with: _emailTextField.text, username: _usernameTextField.text, password: _passwordTextField.text)
+    }
+    
+    @IBAction func getUserActionHandler(_ sender: Any) {
+        presenter.didPressGet()
+    }
+    
+    @IBAction func updateUserActionHandler(_ sender: Any) {
+        presenter.didPressUpdate(email: _emailTextField.text, username: _usernameTextField.text)
+    }
+    
+    @IBAction func deleteUserActionHandler(_ sender: Any) {
+        presenter.didPressDelete()
+    }
+    
+}
+
+// MARK: - Extensions -
+
+extension NetworkingJapxViewController: NetworkingJapxViewInterface {
+    
+    func updateView(with state: NetworkingJapx.State) {
+        _createUserButton.setEnabledAndAlpha(with: state == .UserDoesNotExist)
+        _getUserButton.setEnabledAndAlpha(with: state == .UserExists)
+        _updateUserButton.setEnabledAndAlpha(with: state == .UserExists)
+        _deleteUserButton.setEnabledAndAlpha(with: state == .UserExists)
+    }
+
+}
+
+extension NetworkingJapxViewController: Catalogizable {
+    
+    static var title: String {
+        return "Japx Networking"
+    }
+    
+    static var viewController: UIViewController {
+        return NetworkingJapxWireframe().viewController
+    }
+    
+}
+
+private extension UIButton {
+    
+    func setEnabledAndAlpha(with shouldEnable: Bool) {
+        self.isEnabled = shouldEnable
+        self.alpha = shouldEnable ? 1 : 0.3
+    }
+}
+
+// MARK: - Keyboard notifications -
+
+private extension NetworkingJapxViewController {
+    
+    private func _startObservingKeyboardEvents() {
```

Comment:
If extension is private then all methods are private by default so you can omit `private`